File: 13.调整数组顺序使奇数位于偶数前面.cpp
class Solution {
public:
    void reOrderArray(vector<int> &array) {
        int length = array.size();
        vector<int> helper(array);
        int j = 0;
        for(int i = 0; i < length; i++){
            if(helper[i] % 2)
                array[j++] = helper[i];
        }
        j = length - 1;
        for(int i = length - 1; i >= 0; i--)
            if(helper[i] % 2 == 0)
                array[j--] = helper[i];
    }
};

File: java/13.调整数组顺序使奇数位于偶数前面.java
public class Solution {
    public void reOrderArray(int [] array) {
        if(array == null || array.length == 0)
            return;
        int j = 0;
        int[] arr = new int[array.length];
        for(int i = 0; i < array.length; i++){
            if (array[i] % 2 == 1)
                arr[j++] = (array[i]);
        }
        for(int i = 0; i < array.length; i++){
            if (array[i] % 2 == 0)
                arr[j++] = (array[i]);
        }
        for(int i = 0; i < array.length; i++)
            array[i] = arr[i];
    }
}
